tools/test-case-generators/genall.go

```diff
@@ -0,0 +1,215 @@
+		if err != nil {
+			fmt.Printf("%q failed: %s\n", filename, err)
+			return
+		}
+		saveManifest(manifest, filename)
+		fmt.Printf("Finished job %s\n", filename)
+	}
+	return job, workerName
+}
+
+type DistroArchRepoMap map[string]map[string][]rpmmd.RepoConfig
+
+func readRepos() DistroArchRepoMap {
+	file := "./tools/test-case-generators/repos.json"
+	darm := new(DistroArchRepoMap)
+	fp, err := os.Open(file)
+	if err != nil {
+		panic(err)
+	}
+	defer fp.Close()
+	data, err := ioutil.ReadAll(fp)
+	if err != nil {
+		panic(err)
+	}
+	if err := json.Unmarshal(data, darm); err != nil {
+		panic(err)
+	}
+	return *darm
+}
+
+func depsolve(cacheDir string, packageSets map[string]rpmmd.PackageSet, repos []rpmmd.RepoConfig, d distro.Distro, arch string) (map[string][]rpmmd.PackageSpec, error) {
+	solver := dnfjson.NewSolver(d.ModulePlatformID(), d.Releasever(), arch, cacheDir)
+	packageSpecSets := make(map[string][]rpmmd.PackageSpec)
+	for name, packages := range packageSets {
+		results, err := solver.Depsolve([]rpmmd.PackageSet{packages}, [][]rpmmd.RepoConfig{repos})
+		if err != nil {
+			fmt.Printf("Could not depsolve: %s", err.Error())
+			return nil, err
+		}
+
+		// NOTE: we only depsolve one package set at a time for now
+		result := results[0]
+		packageSpecSets[name] = result.Dependencies
+	}
+	return packageSpecSets, nil
+}
+
+func saveManifest(manifest distro.Manifest, filename string) {
+	b, err := json.MarshalIndent(manifest, "", "  ")
+	if err != nil {
+		panic(err)
+	}
+	fp, err := os.Create("test/data/manifests.noinfo/" + filename)
+	if err != nil {
+		panic(err)
+	}
+	defer fp.Close()
+	fp.Write(b)
+}
+
+type Worker struct {
+	name string
+	done chan bool
+}
+
+func (w *Worker) runJobs(q chan func()) {
+	c := 0
+	go func() {
+		for j := range q {
+			j()
+			c++
+			fmt.Printf("Worker %s finished %d jobs\n", w.name, c)
+		}
+		w.done <- true
+	}()
+}
+
+func (w *Worker) addJob(j func(), q chan func()) {
+	q <- j
+}
+
+func initWorkers(names []string) map[string]*Worker {
+	workers := make(map[string]*Worker)
+	for _, name := range names {
+		// run through once to add one entry per name
+		workers[name] = nil
+	}
+	for name := range workers {
+		w := new(Worker)
+		w.name = name
+		w.done = make(chan bool, 1)
+		workers[name] = w
+	}
+	return workers
+}
+func initWorkersN(n int) map[int]*Worker {
+	workers := make(map[int]*Worker)
+	for idx := 0; idx < n; idx++ {
+		w := new(Worker)
+		name := fmt.Sprint(idx)
+		w.name = name
+		w.done = make(chan bool, 1)
+		workers[idx] = w
+	}
+	return workers
+}
+
+func main() {
+	seedArg := int64(0)
+	darm := readRepos()
+	distros := distroregistry.NewDefault()
+	wNames := make([]string, 0)
+	jobs := make([]func(), 0)
+	fmt.Println("Collecting jobs")
+	for _, distroName := range distros.List() {
+		distribution := distros.GetDistro(distroName)
+		for _, archName := range distribution.ListArches() {
+			if archName == "s390x" {
+				fmt.Println("Skipping s390x")
+				continue
+			}
```

Comment:
Why would you skip `s390x`? 🤔

Reply:
Similar to the installers.  There was some issue with `s390x` that I didn't want to troubleshoot at the time when I was writing this.  It was initially just a speed test / benchmark, so I was skipping anything that was causing issues.

Reply:
I think that it might have been the `tar` image on `rhel-84`... https://github.com/osbuild/osbuild-composer/pull/2350/commits/ef619dbc9b14696713407c89f1be0b50511f9b2b